src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -115,8 +116,34 @@ protected override Task UpdateDisplayAsync(in RenderBatch batch)
             //       buffer on every render.
             var batchBytes = MessagePackSerializer.Serialize(batch, RenderBatchFormatterResolver.Instance);
 
+            if (_client.Connected)
+            {
+                return WriteBatchBytes(batchBytes);
```

Comment:
Does this need to be atomic? Another thread could change the state after `_client.Connected` is evaluated

Reply:
We could attempt to cancel writes on the underlying client if the value changes in interim. In addition, possibly stashing the bytes if we knew the write failed due to client disconnect. Beyond that, I'm not sure what it means to make this atomic.

Reply:
It's not atomic because a another thread can change the state while `WriteBatchBytes` is running. If you read this code and assume a like thread you'd assume that the client always stays connected while `WriteBatchBytes` is executing. That's what I mean by *atomic*.